extensions/positron-assistant/src/models.ts

```diff
@@ -97,6 +105,25 @@ abstract class AILanguageModel implements positron.ai.LanguageModelChatProvider
 		this.provider = _config.provider;
 	}
 
+	async resolveConnection(token: vscode.CancellationToken): Promise<boolean> {
+		token.onCancellationRequested(() => {
+			return false;
+		});
+
+		try {
+			// send a test message to the model
+			await ai.generateText({
+				model: this.model,
+				prompt: 'Hello!',
+			});
+
+			// if the model responds, the config works
+			return true;
+		} catch (error) {
+			return false;
```

Comment:
Can we read this error and show it to the user so they have some idea why the connection failed? Could be a network issue, bad API key, etc.

Reply:
`ai-sdk` returns an APICallError that has some details from Anthropic about what happened. I haven't checked other AI's but this seems to be as granular as it gets with the API call. I could return a more detail from the response like the status code that might be useful for an admin but likely not useful for a user.

![image](https://github.com/user-attachments/assets/ec19b3bc-7e6b-43df-89c2-0424ff40012a)